### PR TITLE
SONAR-6278 Stop relying on graph to store tests

### PR DESCRIPTION
**Pull request by henryju**

### CODE REVIEW

File: sonar-batch/src/main/java/org/sonar/batch/issue/tracking/ServerIssueRepository.java [hunk 1]
```diff
@@ -96,7 +96,7 @@ public Void apply(@Nullable ServerIssue issue) {
     profiler.stopDebug();
   }
 
-  public Iterable<ServerIssue> byComponent(BatchResource component) {
+  public Iterable<ServerIssue> byComponent(BatchComponent component) {
```
Comment (SonarTech) [inline sonar-batch/src/main/java/org/sonar/batch/issue/tracking/ServerIssueRepository.java]: MAJORThe Cyclomatic Complexity of this method "byComponent" is 11 which is greater than 10 authorized.[rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

File: sonar-plugin-api/src/main/java/org/sonar/api/component/Perspective.java [hunk 2]
```diff
@@ -19,6 +19,9 @@
  */
 package org.sonar.api.component;
 
+/**
+ * @deprecated since 5.2
+ */
+@Deprecated
 public interface Perspective<C extends Component> {
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/component/Perspective.java]: NewMAJORC is not used in the interface.[rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2326)
NewINFODo not forget to remove this deprecated code someday.[rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)

File: sonar-batch/src/main/java/org/sonar/batch/test/DefaultTestCase.java [hunk 3]
```diff
@@ -70,100 +70,95 @@ public MutableTestCase setDurationInMs(@Nullable Long l) {
...
   public Status status() {
-    return Status.of((String) getProperty(STATUS));
+    return status;
   }
 
   @Override
   public MutableTestCase setStatus(@Nullable Status s) {
-    setProperty(STATUS, s == null ? null : s.name());
+    this.status = s;
+    ;
```
Comment (SonarTech) [inline sonar-batch/src/main/java/org/sonar/batch/test/DefaultTestCase.java]: NewMAJORRemove this empty statement.[rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AEmptyStatementUsageCheck)

File: sonar-plugin-api/src/main/java/org/sonar/api/component/ResourcePerspectives.java [hunk 4]
```diff
@@ -57,8 +57,10 @@
  * @see Testable
  * @see TestPlan
  * @since 3.5
+ * @deprecated since 5.2. Use appropriate methods on {@link SensorContext}.
  */
-public interface ResourcePerspectives extends Perspectives {
+@Deprecated
+public interface ResourcePerspectives {
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/component/ResourcePerspectives.java]: NewINFODo not forget to remove this deprecated code someday.[rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)

File: sonar-plugin-api/src/main/java/org/sonar/api/test/MutableTestable.java [hunk 5]
```diff
@@ -19,8 +19,6 @@
  */
 package org.sonar.api.test;
 
-import org.sonar.api.component.MutablePerspective;
-
-public interface MutableTestable extends Testable, MutablePerspective {
+public interface MutableTestable extends Testable {
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/test/MutableTestable.java]: NewMAJOR"Testable" is deprecated, implement the suggested replacement instead.[rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1724)

File: sonar-plugin-api/src/main/java/org/sonar/api/test/MutableTestPlan.java [hunk 6]
```diff
@@ -19,15 +19,13 @@
  */
 package org.sonar.api.test;
 
-import org.sonar.api.component.MutablePerspective;
-
-public interface MutableTestPlan extends TestPlan<MutableTestCase>, MutablePerspective {
+public interface MutableTestPlan extends TestPlan<MutableTestCase> {
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/test/MutableTestPlan.java]: NewMAJOR"TestPlan" is deprecated, implement the suggested replacement instead.[rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1724)

File: sonar-batch/src/main/java/org/sonar/batch/test/TestPlanBuilder.java [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+  public TestPlanBuilder() {
+    super(MutableTestPlan.class);
+  }
+
+  @CheckForNull
+  @Override
+  public MutableTestPlan loadPerspective(Class<MutableTestPlan> perspectiveClass, BatchComponent component) {
+    if (component.isFile()) {
+      InputFile inputFile = (InputFile) component.inputPath();
+      if (inputFile.type() == Type.TEST) {
```
Comment (SonarTech) [inline sonar-batch/src/main/java/org/sonar/batch/test/TestPlanBuilder.java]: NewCRITICALPossible null pointer dereference in org.sonar.batch.test.TestPlanBuilder.loadPerspective(Class, BatchComponent) due to return value of called method[rule](http://nemo.sonarqube.org/coding_rules#rule_key=findbugs%3ANP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE)

File: sonar-plugin-api/src/main/java/org/sonar/api/test/TestPlan.java [hunk 8]
```diff
@@ -22,8 +22,9 @@
 import org.sonar.api.component.Perspective;
 
 /**
- * A {@link TestPlan} is 
+ * @deprecated since 5.2 not used
  */
+@Deprecated
 public interface TestPlan<T extends TestCase> extends Perspective {
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/test/TestPlan.java]: NewINFODo not forget to remove this deprecated code someday.[rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)

File: sonar-batch/src/main/java/org/sonar/batch/sensor/DefaultSensorStorage.java [hunk 9]
```diff
@@ -76,12 +76,12 @@
   private final DefaultIndex sonarIndex;
   private final CoverageExclusions coverageExclusions;
   private final DuplicationCache duplicationCache;
-  private final ResourceCache resourceCache;
+  private final BatchComponentCache resourceCache;
   private final ReportPublisher reportPublisher;
 
   public DefaultSensorStorage(MetricFinder metricFinder, Project project, ModuleIssues moduleIssues,
```
Comment (SonarTech) [inline sonar-batch/src/main/java/org/sonar/batch/sensor/DefaultSensorStorage.java]: MAJORMethod has 11 parameters, which is greater than 7 authorized.[rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00107)

File: sonar-batch/src/main/java/org/sonar/batch/issue/tracking/LocalIssueTracking.java [hunk 10]
```diff
@@ -62,12 +62,12 @@
...
   private final ActiveRules activeRules;
   private final InputPathCache inputPathCache;
-  private final ResourceCache resourceCache;
+  private final BatchComponentCache resourceCache;
   private final ServerIssueRepository serverIssueRepository;
   private final ProjectRepositories projectRepositories;
   private final AnalysisMode analysisMode;
 
-  public LocalIssueTracking(ResourceCache resourceCache, IssueCache issueCache, IssueTracking tracking,
+  public LocalIssueTracking(BatchComponentCache resourceCache, IssueCache issueCache, IssueTracking tracking,
```
Comment (SonarTech) [inline sonar-batch/src/main/java/org/sonar/batch/issue/tracking/LocalIssueTracking.java]: MAJORMethod has 11 parameters, which is greater than 7 authorized.[rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00107)

File: sonar-plugin-api/src/main/java/org/sonar/api/test/MutableTestCase.java [hunk 11]
```diff
@@ -38,5 +38,11 @@
   @Deprecated
   MutableTestCase setType(@Nullable String s);
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/test/MutableTestCase.java]: INFODo not forget to remove this deprecated code someday.[rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)

File: sonar-plugin-api/src/main/java/org/sonar/api/test/MutableTestCase.java [hunk 12]
```diff
@@ -19,9 +19,9 @@
  */
 package org.sonar.api.test;
 
-import javax.annotation.Nullable;
-
 import java.util.List;
+import javax.annotation.Nullable;
+import org.sonar.api.batch.fs.InputFile;
 
 public interface MutableTestCase extends TestCase {
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/test/MutableTestCase.java]: NewMAJOR"TestCase" is deprecated, implement the suggested replacement instead.[rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1724)

File: sonar-plugin-api/src/main/java/org/sonar/api/test/MutableTestCase.java [hunk 13]
```diff
@@ -38,5 +38,11 @@
   @Deprecated
   MutableTestCase setType(@Nullable String s);
 
+  /**
+   * @deprecated since 5.2. Use {@link #setCoverageBlock(InputFile, List)}
+   */
+  @Deprecated
   MutableTestCase setCoverageBlock(Testable testable, List<Integer> lines);
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/test/MutableTestCase.java]: NewINFODo not forget to remove this deprecated code someday.[rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)

File: sonar-plugin-api/src/main/java/org/sonar/api/component/Component.java [hunk 14]
```diff
@@ -21,6 +21,10 @@
 
 import javax.annotation.CheckForNull;
 
+/**
+ * @deprecated since 5.2
+ */
+@Deprecated
 public interface Component {
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/component/Component.java]: NewINFODo not forget to remove this deprecated code someday.[rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)

File: sonar-plugin-api/src/main/java/org/sonar/api/test/TestCase.java [hunk 15]
```diff
@@ -23,8 +23,9 @@
 import javax.annotation.Nullable;
 
 /**
- * Represents a simple test execution result.
+ * @deprecated since 5.2 not used
  */
+@Deprecated
 public interface TestCase {
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/test/TestCase.java]: NewINFODo not forget to remove this deprecated code someday.[rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)

File: sonar-batch/src/main/java/org/sonar/batch/source/HighlightableBuilder.java [hunk 16]
```diff
@@ -19,46 +19,29 @@
...
-      if (batchComponent != null) {
-        InputFile path = (InputFile) batchComponent.inputPath();
-        if (path != null) {
-          return new DefaultHighlightable((DefaultInputFile) path, sensorStorage);
-        }
-      }
+  public Highlightable loadPerspective(Class<Highlightable> perspectiveClass, BatchComponent component) {
+    if (component.isFile()) {
+      InputFile path = (InputFile) component.inputPath();
+      return new DefaultHighlightable((DefaultInputFile) path, sensorStorage);
```
Comment (SonarTech) [inline sonar-batch/src/main/java/org/sonar/batch/source/HighlightableBuilder.java]: NewCRITICALPossible null pointer dereference in org.sonar.batch.source.HighlightableBuilder.loadPerspective(Class, BatchComponent) due to return value of called method[rule](http://nemo.sonarqube.org/coding_rules#rule_key=findbugs%3ANP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE)

File: sonar-batch/src/main/java/org/sonar/batch/report/TestExecutionAndCoveragePublisher.java [hunk 17]
```diff
@@ -105,36 +104,36 @@ public CoverageDetail apply(@Nonnull String testName) {
...
+  public TestExecutionAndCoveragePublisher(BatchComponentCache resourceCache, TestPlanBuilder testPlanBuilder) {
+    this.componentCache = resourceCache;
     this.testPlanBuilder = testPlanBuilder;
   }
 
   @Override
   public void publish(BatchReportWriter writer) {
-    for (final BatchResource resource : resourceCache.all()) {
-      if (!resource.isFile()) {
+    for (final BatchComponent component : componentCache.all()) {
```
Comment (SonarTech) [inline sonar-batch/src/main/java/org/sonar/batch/report/TestExecutionAndCoveragePublisher.java]: MAJORReduce the total number of break and continue statements in this loop to use at most one.[rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS135)

File: sonar-batch/src/main/java/org/sonar/batch/report/TestExecutionAndCoveragePublisher.java [hunk 18]
```diff
@@ -95,7 +94,7 @@ public CoverageDetail apply(@Nonnull String testName) {
       builder.setTestName(testName);
       for (CoverageBlock block : testCase.coverageBlocks()) {
         coveredBuilder.clear();
-        coveredBuilder.setFileRef(resourceCache.get(block.testable().component().key()).batchId());
+        coveredBuilder.setFileRef(componentCache.get(((DefaultTestable) block.testable()).inputFile().key()).batchId());
```
Comment (SonarTech) [inline sonar-batch/src/main/java/org/sonar/batch/report/TestExecutionAndCoveragePublisher.java]: NewBLOCKERNullPointerException might be thrown as 'get' is nullable here[rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2259)

File: sonar-plugin-api/src/main/java/org/sonar/api/test/Testable.java [hunk 19]
```diff
@@ -19,12 +19,15 @@
...
 import java.util.List;
 import java.util.Map;
 import java.util.SortedSet;
+import org.sonar.api.component.Perspective;
 
+/**
+ * @deprecated since 5.2
+ */
+@Deprecated
 public interface Testable extends Perspective {
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/test/Testable.java]: NewINFODo not forget to remove this deprecated code someday.[rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)

File: sonar-batch/src/main/java/org/sonar/batch/test/TestableBuilder.java [hunk 20]
```diff
@@ -17,22 +17,31 @@
...
   }
 
+  @CheckForNull
   @Override
-  protected Class<? extends BeanVertex> getBeanClass() {
-    return DefaultTestable.class;
+  public MutableTestable loadPerspective(Class<MutableTestable> perspectiveClass, BatchComponent component) {
+    if (component.isFile()) {
+      InputFile inputFile = (InputFile) component.inputPath();
+      if (inputFile.type() == Type.MAIN) {
```
Comment (SonarTech) [inline sonar-batch/src/main/java/org/sonar/batch/test/TestableBuilder.java]: NewCRITICALPossible null pointer dereference in org.sonar.batch.test.TestableBuilder.loadPerspective(Class, BatchComponent) due to return value of called method[rule](http://nemo.sonarqube.org/coding_rules#rule_key=findbugs%3ANP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE)

File: sonar-batch/src/main/java/org/sonar/batch/source/SymbolizableBuilder.java [hunk 21]
```diff
@@ -20,46 +20,29 @@
...
-      if (batchComponent != null) {
-        InputFile path = (InputFile) batchComponent.inputPath();
-        if (path != null) {
-          return new DefaultSymbolizable((DefaultInputFile) path, sensorStorage);
-        }
-      }
+  public Symbolizable loadPerspective(Class<Symbolizable> perspectiveClass, BatchComponent component) {
+    if (component.isFile()) {
+      InputFile path = (InputFile) component.inputPath();
+      return new DefaultSymbolizable((DefaultInputFile) path, sensorStorage);
```
Comment (SonarTech) [inline sonar-batch/src/main/java/org/sonar/batch/source/SymbolizableBuilder.java]: NewCRITICALPossible null pointer dereference in org.sonar.batch.source.SymbolizableBuilder.loadPerspective(Class, BatchComponent) due to return value of called method[rule](http://nemo.sonarqube.org/coding_rules#rule_key=findbugs%3ANP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE)

**Comment by SonarTech** [comment]
SonarQube analysis reported 23 new issues:
* 1 blocker
* 9 critical
* 6 major
* 7 info

Details:
* [Remove this unused "r" local variable.](https://github.com/SonarSource/sonarqube/blob/c8549c2bebd024e17c17ea436b17fd23ec4d7ff8/sonar-batch/src/main/java/org/sonar/batch/deprecated/perspectives/BatchPerspectives.java#L66)[rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1481)
* [NullPointerException might be thrown as 'get' is nullable here](https://github.com/SonarSource/sonarqube/blob/c8549c2bebd024e17c17ea436b17fd23ec4d7ff8/sonar-batch/src/main/java/org/sonar/batch/report/TestExecutionAndCoveragePublisher.java#L97)[rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2259)
* [Possible null pointer dereference in org.sonar.batch.source.HighlightableBuilder.loadPerspective(Class, BatchComponent) due to return value of called method](https://github.com/SonarSource/sonarqube/blob/c8549c2bebd024e17c17ea436b17fd23ec4d7ff8/sonar-batch/src/main/java/org/sonar/batch/source/HighlightableBuilder.java#L44)[rule](http://nemo.sonarqube.org/coding_rules#rule_key=findbugs%3ANP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE)
* [Possible null pointer dereference in org.sonar.batch.source.SymbolizableBuilder.loadPerspective(Class, BatchComponent) due to return value of called method](https://github.com/SonarSource/sonarqube/blob/c8549c2bebd024e17c17ea436b17fd23ec4d7ff8/sonar-batch/src/main/java/org/sonar/batch/source/SymbolizableBuilder.java#L45)[rule](http://nemo.sonarqube.org/coding_rules#rule_key=findbugs%3ANP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE)
* [Remove this empty statement.](https://github.com/SonarSource/sonarqube/blob/c8549c2bebd024e17c17ea436b17fd23ec4d7ff8/sonar-batch/src/main/java/org/sonar/batch/test/DefaultTestCase.java#L85)[rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AEmptyStatementUsageCheck)
* [Possible null pointer dereference in org.sonar.batch.test.TestPlanBuilder.loadPerspective(Class, BatchComponent) due to return value of called method](https://github.com/SonarSource/sonarqube/blob/c8549c2bebd024e17c17ea436b17fd23ec4d7ff8/sonar-batch/src/main/java/org/sonar/batch/test/TestPlanBuilder.java#L44)[rule](http://nemo.sonarqube.org/coding_rules#rule_key=findbugs%3ANP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE)
* [Possible null pointer dereference in org.sonar.batch.test.TestableBuilder.loadPerspective(Class, BatchComponent) due to return value of called method](https://github.com/SonarSource/sonarqube/blob/c8549c2bebd024e17c17ea436b17fd23ec4d7ff8/sonar-batch/src/main/java/org/sonar/batch/test/TestableBuilder.java#L41)[rule](http://nemo.sonarqube.org/coding_rules#rule_key=findbugs%3ANP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE)
* [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/c8549c2bebd024e17c17ea436b17fd23ec4d7ff8/sonar-plugin-api/src/main/java/org/sonar/api/component/Component.java#L28)[rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* [Method 'public org.sonar.api.component.Component component()' has been removed](https://github.com/SonarSource/sonarqube/blob/c8549c2bebd024e17c17ea436b17fd23ec4d7ff8/sonar-plugin-api/src/main/java/org/sonar/api/component/Perspective.java)[rule](http://nemo.sonarqube.org/coding_rules#rule_key=clirr%3Aclirr-api-break)
* [C is not used in the interface.](https://github.com/SonarSource/sonarqube/blob/c8549c2bebd024e17c17ea436b17fd23ec4d7ff8/sonar-plugin-api/src/main/java/org/sonar/api/component/Perspective.java#L26)[rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2326)
* [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/c8549c2bebd024e17c17ea436b17fd23ec4d7ff8/sonar-plugin-api/src/main/java/org/sonar/api/component/Perspective.java#L26)[rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/c8549c2bebd024e17c17ea436b17fd23ec4d7ff8/sonar-plugin-api/src/main/java/org/sonar/api/component/ResourcePerspectives.java#L63)[rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* [Removed org.sonar.api.BatchComponent from the set of implemented interfaces](https://github.com/SonarSource/sonarqube/blob/c8549c2bebd024e17c17ea436b17fd23ec4d7ff8/sonar-plugin-api/src/main/java/org/sonar/api/component/ResourcePerspectives.java)[rule](http://nemo.sonarqube.org/coding_rules#rule_key=clirr%3Aclirr-api-break)
* ["TestCase" is deprecated, implement the suggested replacement instead.](https://github.com/SonarSource/sonarqube/blob/c8549c2bebd024e17c17ea436b17fd23ec4d7ff8/sonar-plugin-api/src/main/java/org/sonar/api/test/MutableTestCase.java#L26)[rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1724)
* [Method 'public org.sonar.api.test.MutableTestCase setCoverageBlock(org.sonar.api.batch.fs.InputFile, java.util.List)' has been added to an interface](https://github.com/SonarSource/sonarqube/blob/c8549c2bebd024e17c17ea436b17fd23ec4d7ff8/sonar-plugin-api/src/main/java/org/sonar/api/test/MutableTestCase.java)[rule](http://nemo.sonarqube.org/coding_rules#rule_key=clirr%3Aclirr-api-break)
* [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/c8549c2bebd024e17c17ea436b17fd23ec4d7ff8/sonar-plugin-api/src/main/java/org/sonar/api/test/MutableTestCase.java#L45)[rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* ["TestPlan" is deprecated, implement the suggested replacement instead.](https://github.com/SonarSource/sonarqube/blob/c8549c2bebd024e17c17ea436b17fd23ec4d7ff8/sonar-plugin-api/src/main/java/org/sonar/api/test/MutableTestPlan.java#L22)[rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1724)
* [Removed org.sonar.api.component.MutablePerspective from the set of implemented interfaces](https://github.com/SonarSource/sonarqube/blob/c8549c2bebd024e17c17ea436b17fd23ec4d7ff8/sonar-plugin-api/src/main/java/org/sonar/api/test/MutableTestPlan.java)[rule](http://nemo.sonarqube.org/coding_rules#rule_key=clirr%3Aclirr-api-break)
* [Removed org.sonar.api.component.MutablePerspective from the set of implemented interfaces](https://github.com/SonarSource/sonarqube/blob/c8549c2bebd024e17c17ea436b17fd23ec4d7ff8/sonar-plugin-api/src/main/java/org/sonar/api/test/MutableTestable.java)[rule](http://nemo.sonarqube.org/coding_rules#rule_key=clirr%3Aclirr-api-break)
* ["Testable" is deprecated, implement the suggested replacement instead.](https://github.com/SonarSource/sonarqube/blob/c8549c2bebd024e17c17ea436b17fd23ec4d7ff8/sonar-plugin-api/src/main/java/org/sonar/api/test/MutableTestable.java#L22)[rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1724)
* [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/c8549c2bebd024e17c17ea436b17fd23ec4d7ff8/sonar-plugin-api/src/main/java/org/sonar/api/test/TestCase.java#L29)[rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/c8549c2bebd024e17c17ea436b17fd23ec4d7ff8/sonar-plugin-api/src/main/java/org/sonar/api/test/TestPlan.java#L28)[rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/c8549c2bebd024e17c17ea436b17fd23ec4d7ff8/sonar-plugin-api/src/main/java/org/sonar/api/test/Testable.java#L31)[rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)

**Comment by simonbrandhof** [comment]
PR could have been split into 2 commits: one for renaming of ResourceCache and one from removal of graph representation of tests. It would have removed the noise.
LGTM

**Pull request closed without merging**